packages/app/src/components/VisualizationOptions/Options/ApprovalLevel.js

```diff
@@ -0,0 +1,153 @@
+import React, { useState } from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { createSelector } from 'reselect'
+
+import i18n from '@dhis2/d2-i18n'
+
+import { SingleSelectField, SingleSelectOption } from '@dhis2/ui-core'
+import { useDataEngine } from '@dhis2/app-runtime'
+
+import { sGetUserAuthorities } from '../../../reducers/user'
+import { sGetSettings } from '../../../reducers/settings'
+import { sGetUiOptions } from '../../../reducers/ui'
+import { acSetUiOptions } from '../../../actions/ui'
+
+export const APPROVAL_LEVEL_OPTION_AUTH = 'F_VIEW_UNAPPROVED_DATA'
+
+const ApprovalLevelSelect = ({
+    value,
+    loading,
+    options,
+    onFocus,
+    onChange,
+}) => {
+    const selected =
+        value && value.id ? { value: value.id, label: value.displayName } : {}
+
+    return (
+        <SingleSelectField
+            name="approvalLevelSelect"
+            label={i18n.t('Data approved at level')}
+            selected={selected}
+            inputWidth="280px"
+            placeholder={i18n.t('Select from predefined levels')}
+            loadingText={i18n.t('Loading data approval levels')}
+            loading={loading}
+            dense
+            onFocus={onFocus}
+            onChange={({ selected }) =>
+                onChange({
+                    id: selected.value,
+                    displayName: selected.label,
+                })
+            }
+        >
+            {options.map(({ value, label }) => (
+                <SingleSelectOption key={value} value={value} label={label} />
+            ))}
+        </SingleSelectField>
+    )
+}
+
+ApprovalLevelSelect.propTypes = {
+    loading: PropTypes.bool,
+    options: PropTypes.array,
+    value: PropTypes.object,
+    onChange: PropTypes.func,
+    onFocus: PropTypes.func,
+}
+
+const ApprovalLevel = ({ value, onChange, enabled }) => {
+    const engine = useDataEngine()
+
+    const [options, setOptions] = useState([])
+    const [isLoaded, setIsLoaded] = useState(false)
+
+    if (value && value.id) {
+        if (!options.find(option => option.value === value.id)) {
+            setOptions([
+                ...options,
+                { value: value.id, label: value.displayName },
+            ])
+        }
+    }
+
+    if (!enabled) {
+        return null
+    }
+
+    if (value && value.id) {
+        if ((!options, find(option => option.value === value.id))) {
+            setOptions([
+                ...options,
+                { value: value.id, label: value.displayName },
+            ])
+        }
+    }
+
+    const onSelectFocus = async () => {
+        if (!isLoaded) {
+            const query = {
```

Comment:
Approving this for now, but imho we should add a separate PR to move this query to `src/api` and make it static. Together with this as well https://github.com/dhis2/data-visualizer-app/blob/b602425c10d779b44a4db7d679b521c85e445fa3/packages/app/src/components/VisualizationOptions/Options/LegendSet.js#L83

Reply:
Moved the query constant at the top as per @amcgee suggestion.